app/src/lib/shells/linux.ts

```diff
@@ -1,26 +1,81 @@
 import { spawn } from 'child_process'
+import * as fs from 'fs'
+import { assertNever, fatalError } from '../fatal-error'
 import { IFoundShell } from './found-shell'
 
 export enum Shell {
   Gnome = 'gnome-terminal',
+  Tilix = 'tilix',
 }
 
 export const Default = Shell.Gnome
 
 export function parse(label: string): Shell {
+  if (label === Shell.Gnome) {
+    return Shell.Gnome
+  }
+
+  if (label === Shell.Tilix) {
+    return Shell.Tilix
+  }
+
   return Default
 }
 
+async function getPathIfAvailable(path: string): Promise<string | null> {
+  return new Promise<string | null>(resolve => {
+    fs.stat(path, err => {
+      if (err) {
+        resolve(null)
+      } else {
+        resolve(path)
+      }
+    })
+  })
+}
+
+async function getShellPath(shell: Shell): Promise<string | null> {
+  switch (shell) {
+    case Shell.Gnome:
+      return await getPathIfAvailable('/usr/bin/gnome-terminal')
```

Comment:
Given we don't have any try-catch code here, we _could_ just make this return the promise directly, which would then make the `async` keyword unnecessary.